components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -470,6 +484,19 @@ void BraveProfileSyncServiceImpl::OnSyncReady() {
   DCHECK(false == brave_sync_ready_);
   brave_sync_ready_ = true;
 
+  DCHECK(model_);
+  if (model_->loaded()) {
+    OnSyncReadyBookmarksModelLoaded();
+  } else {
+    // Will call OnSyncReadyBookmarksModelLoaded once model is loaded
+    LOG(WARNING) << "[BraveSync] bookmarks model is not yet loaded, "
+                 << "OnSyncReady will be delayed";
+    model_->AddObserver(this);
```

Comment:
why aren't we just adding the observer in the constructor?

Reply:
because usually model is loaded before the sync service starts. in fact I had to put the `wait` into STR to be able to reproduce it. So observer is set when it is required and observer is not set when we are good without it.